src/adapter/services/LegacyPersistenceAdapter.js

```diff
@@ -43,6 +43,10 @@ export default class LegacyPersistenceAdapter {
             key: key
         };
 
-        return this.openmct.legacyObject(this.openmct.objects.get(identifier));
+        return this.openmct.objects.get(identifier).then(domainObject => {
+            let object = this.openmct.legacyObject(domainObject);
+
+            return object.model;
```

Comment:
we should be returning the entire object ? Because any code using legacy objects expect a full legacy object not just the model

Reply:
This breaks if we don't return object.model as it is looking for `domainObject.composition`
https://github.com/nasa/openmct/blob/master/platform/search/src/services/GenericSearchProvider.js#L192